core/app/models/spree/weighted_adjustments_handler.rb

```diff
@@ -0,0 +1,47 @@
+module Spree
+  class WeightedAdjustmentsHandler
```

Comment:
Something similar to whatever you choose for the calculator name seems fine imo.  Should this go in the `models/spree/promotion/` folder somewhere though?